tasks/Debian.yml

```diff
@@ -1,15 +1,26 @@
----
-- name: Add specific key
-  apt_key: id={{docker_repo_key}} keyserver=keyserver.ubuntu.com state=present
 
-- name: Install https apt transport package
-  apt: pkg=apt-transport-https state=present update_cache=yes
+- name: Add specific key
+  apt_key: keyserver={{docker_repo_key_server}} id={{docker_repo_key}}
 
 - name: Add docker repo
-  apt_repository: repo='deb {{ docker_repo }} docker main' state=present
+  apt_repository: repo='deb {{docker_repo}} ubuntu-{{ansible_distribution_release}} main' state=present
```

Comment:
One problem I see here is you only expect to be installed on Ubuntu system by using `ubuntu-{{ansible_distribution_release}}`. For vanilla Debian system, it's `debian-{{ansible_distribution_release}}`.